### PR TITLE
Single squashed commit containing all the changes related to making the solidity repository standalone

### PR DESCRIPTION
**Pull request by bobsummerwill**
Replaces https://github.com/ethereum/solidity/pull/698.

The Solidity unit-tests have already been switched to talk to an 'eth' node over IPC, rather than relying on an in-process EVM.

The two other large pieces of the puzzle here were copying of a subset of CMake files into this package, together with the subset of libdevbase needed here.

### CODE REVIEW

**Comment by VoR0220** [comment]
Dammit! Heisenbugged!

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,86 @@
+#------------------------------------------------------------------------------
+# TravisCI configuration file for solidity.
+#
+# The documentation for solidity is hosted at:
+#
+# http://solidity.readthedocs.org
+#
+# TODO - Merge with .travis.yml file which currently lives in the root of
+# webthree-umbrella, but actually contains the automation for the Solidity
```
Comment (chriseth) [inline .travis.yml:9]: Emscripten is our main build target. It **has** to be added before this is merged.

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,86 @@
...
\ No newline at end of file
+cache:
+    - ccache
+    - directories: build
+install:
+    - ./scripts/install_deps.sh
+before_script:
+    - mkdir -p build
+    - cd build
+    - cmake .. -DCMAKE_BUILD_TYPE=$TRAVIS_BUILD_TYPE
```
Comment (chriseth) [inline .travis.yml]: As I said earlier: If we leave it like that, `make` will run even if `cmake` failed. Even worse: `release.sh` will run if make failed.

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,86 @@
...
+    - make -j2
+    - ../scripts/release.sh $ZIP_SUFFIX
+script:
+    - ETH_PATH_FOR_linux="eth"
+    - ETH_PATH_FOR_osx="../eth"
+    - ETH_PATH_EXPRESSION_NAME=ETH_PATH_FOR_$TRAVIS_OS_NAME
+    - ETH_PATH=$ETH_PATH_EXPRESSION_NAME
+    - eval \$$ETH_PATH --test -d /tmp/test &
+    - while [ ! -S /tmp/test/geth.ipc ]; do sleep 2; done
```
Comment (chriseth) [inline .travis.yml]: Please move this into a script. The travis file should be used for configuration, but the actual scripts should be somewhere else. The other benefit of moving this to a script would be that this can be run outside of travis.

File: .travis.yml [hunk 4]
```diff
@@ -1,15 +1,86 @@
...
+    - export ETH_TEST_IPC=/tmp/test/geth.ipc && ./test/soltest
```
Comment (chriseth) [inline .travis.yml]: It should be possible (at least in a shell script) to shorten this to:

```
ETH_TEST_IPC=/tmp/test/geth.ipc ./test/soltest
```

You can also use:

```
./test/soltest --ipc /tmp/test/geth.ipc
```

File: .travis.yml [hunk 5]
```diff
@@ -1,15 +1,86 @@
...
+    - pkill eth
+after_success:
+    - DOCS_COMMAND_FOR_linux="cd docs && sphinx-build -nW -b html -d _build/doctrees . _build/html && cd .."
```
Comment (chriseth) [inline .travis.yml]: This should be a separate matrix job.

File: CMakeLists.txt [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+set(ETH_CMAKE_DIR   "${CMAKE_CURRENT_LIST_DIR}/cmake"   CACHE PATH "The the path to the cmake directory")
 list(APPEND CMAKE_MODULE_PATH ${ETH_CMAKE_DIR})
 
 # Set cmake_policies
 include(EthPolicy)
 eth_policy()
 
 # project name and version should be set after cmake_policy CMP0048
-set(PROJECT_VERSION "0.3.5")
+set(PROJECT_VERSION "0.3.6")
```
Comment (chriseth) [inline CMakeLists.txt]: Let us change the version in its own PR.

File: .travis.yml (hunk 1 above)
Comment (bobsummerwill) [inline .travis.yml:9]: Just about done.
